include/invert_quda.h

```diff
@@ -1198,9 +1198,20 @@ namespace quda {
       ColorSpinorField *out=nullptr;
       ColorSpinorField *in=nullptr;
 
-      dirac.prepare(in, out, x, b, solution_type);
+      bool special_prepare_recon
+        = (dirac.getDiracType() == QUDA_STAGGEREDKD_DIRAC || dirac.getDiracType() == QUDA_ASQTADKD_DIRAC);
+
```

Comment:
Since this is a property of the Dirac operator, why can't this be hidden inside the Dirac operator implementation rather than exposing this if / else condition to all?

Reply:
No good reason, fixed